src/Loop.php

```diff
@@ -33,6 +47,20 @@ public static function execute(callable $callback, LoopDriver $driver)
     }
 
     /**
+     * Create a new driver if a factory is present, otherwise throw.
+     *
+     * @throws \LogicException if no factory is set
+     */
+    private static function createDriver()
+    {
+        if (self::$factory === null) {
+            throw new \LogicException("Can't create an event loop driver without a factory.");
+        }
+
+        return self::$factory->create();
```

Comment:
@kelunik 

> Totally agree, but how does it relate to...

If someone can create a factory that returns a string, other internal code will break horribly. We're in agreement that this is a good idea (from the sounds of things it's already implemented).